### ISSUE TITLE
IndexOutOfBoundsException in applyPattern

### ISSUE THREAD

**Issue by LudovicAns** [Rysefoxx/RyseInventory#29]
**Describe the bug**
Throw IndexOutOfBoundsException when using SlotIteratorPattern in SlotIterator.
```
[16:40:14 WARN]: [TWStaffChat] Task #14 for TWStaffChat v1.0.0 generated an exception
java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
        at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64) ~[?:?]
        at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70) ~[?:?]
        at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266) ~[?:?]
        at java.util.Objects.checkIndex(Objects.java:359) ~[?:?]
        at java.util.ArrayList.get(ArrayList.java:427) ~[?:?]
        at io.github.rysefoxx.pagination.RyseInventory.applyPattern(RyseInventory.java:1626) ~[TerraWildAPI-2.2.0.jar:?]
        at io.github.rysefoxx.pagination.RyseInventory.loadByPage(RyseInventory.java:1496) ~[TerraWildAPI-2.2.0.jar:?]
        at io.github.rysefoxx.pagination.RyseInventory.initInventory(RyseInventory.java:696) ~[TerraWildAPI-2.2.0.jar:?]
        at io.github.rysefoxx.pagination.RyseInventory.lambda$open$8(RyseInventory.java:599) ~[TerraWildAPI-2.2.0.jar:?]
        at org.bukkit.craftbukkit.v1_19_R1.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.19.2.jar:git-Paper-194]
        at org.bukkit.craftbukkit.v1_19_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:483) ~[paper-1.19.2.jar:git-Paper-194]
        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1473) ~[paper-1.19.2.jar:git-Paper-194]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:446) ~[paper-1.19.2.jar:git-Paper-194]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1397) ~[paper-1.19.2.jar:git-Paper-194]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1173) ~[paper-1.19.2.jar:git-Paper-194]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:305) ~[paper-1.19.2.jar:git-Paper-194]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```
**To Reproduce**
1. Create a SlotIterator using SlotIteratorPattern
```
private SlotIterator slotIterator() {
        SlotIteratorPattern slotIteratorPattern = SlotIteratorPattern
                .builder()
                .define("XXXXXXXXX",
                        "XOOOOOOOX",
                        "XOOOOOOOX",
                        "XOOOOOOOX",
                        "XOOOOOOOX",
                        "XXXXXXXXX")
                .attach('O')
                .buildPattern();

        return SlotIterator
                .builder()
                .withPattern(slotIteratorPattern)
                .build();
    }
```
2. Try to open your inventory.

**Expected behavior**
Page filled with ItemStack.

**Code (Optional)**
```
package fr.terrawild.twstaffchat;

import fr.terrawild.terrawildapi.configuration.ConfigurationManager;
import fr.terrawild.terrawildapi.configuration.YAMLParser;
import fr.terrawild.terrawildapi.item.ItemBuilder;
import fr.terrawild.terrawildapi.utils.ColoredText;
import io.github.rysefoxx.SlotIterator;
import io.github.rysefoxx.content.IntelligentItem;
import io.github.rysefoxx.content.InventoryProvider;
import io.github.rysefoxx.pagination.InventoryContents;
import io.github.rysefoxx.pagination.Pagination;
import io.github.rysefoxx.pagination.RyseInventory;
import io.github.rysefoxx.pattern.SlotIteratorPattern;
import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

public class STCInventoryProvider implements InventoryProvider {

    private final ConfigurationManager cm;
    private final RyseInventory ryseInventory;

    public STCInventoryProvider(ConfigurationManager cm, Player player) {
        this.cm = cm;

        ryseInventory = RyseInventory.builder()
                .identifier("STCInventory-" + player.getUniqueId())
                .provider(this)
                .size(6*9)
                .manager(TWStaffChat.getINSTANCE().getTerraWildAPI().getInventoryManager())
                .title("Staff Chat")
                .build(TWStaffChat.getINSTANCE());
    }

    @Override
    public void init(Player player, InventoryContents contents) {
        final FileConfiguration cfg = cm.getConfigurationFile("config.yml");
        final ConfigurationSection messages = cfg.getConfigurationSection("messages");

        if (messages == null) return;

        Pagination pagination = contents.pagination();

        for (String key : messages.getKeys(false)) {
            final ConfigurationSection messageSection = messages.getConfigurationSection(key);
            if (messageSection == null) return;

            final YAMLParser yamlParser = new YAMLParser(cfg);
            ItemStack itemStack = yamlParser.getItemStack(
                    messageSection.getCurrentPath() + ".item",
                    new ItemBuilder(Material.BARREL)
                            .setDisplayName(new ColoredText("&cErreur de configuration de l'item !").buildComponent())
                            .getItemStack()
            );
            assert itemStack != null;

            final String message = messageSection.getString("message");
            if (message == null) {
                pagination.addItem(IntelligentItem.empty(itemStack));
            } else {
                pagination.addItem(IntelligentItem.of(itemStack, e -> {
                    Player whoClicked = (Player) e.getWhoClicked();
                    whoClicked.chat(ChatColor.translateAlternateColorCodes('&', message));
                    whoClicked.closeInventory();
                }));
            }

        }

        pagination.iterator(slotIterator());
    }

    private SlotIterator slotIterator() {
        SlotIteratorPattern slotIteratorPattern = SlotIteratorPattern
                .builder()
                .define("XXXXXXXXX",
                        "XOOOOOOOX",
                        "XOOOOOOOX",
                        "XOOOOOOOX",
                        "XOOOOOOOX",
                        "XXXXXXXXX")
                .attach('O')
                .buildPattern();

        return SlotIterator
                .builder()
                .withPattern(slotIteratorPattern)
                .build();
    }

    public RyseInventory getRyseInventory() {
        return ryseInventory;
    }
}
```

**Comment by Rysefoxx** [comment]
Will be fixed in update 1.5.2.

**Comment by LudovicAns** [comment]
> Will be fixed in update 1.5.2.

Can you make a PRE-RELEASE of this version ? I actively dev with your API ..

**Comment by Rysefoxx** [comment]
> > Will be fixed in update 1.5.2.
> 
> Can you make a PRE-RELEASE of this version ? I actively dev with your API ..

The next update will come when it is ready. Sorry that it takes longer.

**Comment by Rysefoxx** [comment]
[Update 1.5.2](https://github.com/Rysefoxx/RyseInventory/releases/tag/v1.5.2) is out
